packages/app-generic-asset/src/Transfer.tsx

```diff
@@ -54,8 +54,12 @@ function Transfer ({ assets, className, onClose, recipientId: propRecipientId, s
   }, [assets]);
 
   useEffect((): void => {
-    setHasBalance(assetBalance && !assetBalance.isZero());
```

Comment:
Can you explain, why this wasn't behaving as expected?

Reply:
thats cause when we assetBalance && !assetBalance.isZero(), typescript didn't like as TS2345: Argument of type 'boolean | undefined' is not assignable to parameter of type 'SetStateAction<boolean>'.
cause assetBalance can be undefined.. if it is undefined, the other part of && won't be executed ever and undefined would be set as setHasBalance, when I want it to take only boolean..

Reply:
You should have tried Boolean(value), have you? `setHasBalance(Boolean(assetBalance) && !assetBalance.isZero())` looks safer to me

Reply:
tried..it have ts error TS2532: Object is possibly 'undefined' at !assetBalance.isZero()

Reply:
What about assetBalance.isDefined()?

Reply:
I can't add it as a part of setHasBalance.. but can add a condition if it is defined then setHasBalance.. which is what I ended up doing if (assetBalance) { ... }

Reply:
The problem is the behaviour is different in (if...) unless you add the else case too

Reply:
changed as per our discussion..